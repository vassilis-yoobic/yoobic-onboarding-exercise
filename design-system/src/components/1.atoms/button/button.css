button {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  padding: 20px;
  font-weight: 700;
  -webkit-transition: all 0.15s ease 0s;
  transition: all 0.15s ease 0s;
  text-transform: uppercase;
  letter-spacing: 0.56px;
  text-decoration: none;
  border: none;
  outline: none;
  cursor: pointer;
}

button.Primary-Rounded, button.Primary-Square {
  color: #fff;
  background-color: #4c48ff;
}

button.Primary-Rounded:hover, button.Primary-Square:hover {
  background-color: #3d3aff;
}

button.Secondary-Rounded, button.Secondary-Square {
  color: #4c48ff;
  background-color: #f4f4ff;
}

button.Secondary-Rounded:hover, button.Secondary-Square:hover {
  background-color: #fff;
}

button.Primary-Rounded, button.Secondary-Rounded {
  border-radius: 100px;
}

button.Primary-Square, button.Secondary-Square {
  border-radius: 8px;
}

button.Third-Square {
  background-color: #161525;
  color: #e0e0f1;
  border-radius: 8px;
}

button.With-Gradient {
  border-radius: 200px;
  text-transform: lowercase;
  color: #fff;
  background: linear-gradient(89.97deg, #5353ff 0.03%, #ff4a55 99.14%);
}

button.Small {
  font-size: 12px;
  padding: 6px 12px;
}

button.Medium {
  font-size: 14px;
}
/*# sourceMappingURL=button.css.map */