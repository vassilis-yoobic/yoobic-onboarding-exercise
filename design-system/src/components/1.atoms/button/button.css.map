{
    "version": 3,
    "mappings": "AAEA,AAAA,MAAM,CAAC;EACL,OAAO,EAAE,IAAI;EACb,WAAW,EAAE,MAAM;EACnB,OAAO,EAAE,IAAI;EACb,WAAW,EAAE,GAAG;EAChB,UAAU,EAAE,iBAAiB;EAC7B,cAAc,EAAE,SAAS;EACzB,cAAc,EAAE,MAAM;EACtB,eAAe,EAAE,IAAI;EACrB,MAAM,EAAE,IAAI;EACZ,OAAO,EAAE,IAAI;EACb,MAAM,EAAE,OAAO;CAiDhB;;AA5DD,AAaE,MAbI,AAaH,gBAAgB,EAbnB,MAAM,AAcH,eAAe,CAAC;EACb,KAAK,ECXG,IAAI;EDYZ,gBAAgB,ECjBF,OAAO;CDqBxB;;AApBH,AAiBM,MAjBA,AAaH,gBAAgB,AAIZ,MAAM,EAjBb,MAAM,AAcH,eAAe,AAGX,MAAM,CAAC;EACN,gBAAgB,EClBH,OAAO;CDmBrB;;AAnBP,AAqBE,MArBI,AAqBH,kBAAkB,EArBrB,MAAM,AAsBH,iBAAiB,CAAC;EACf,KAAK,ECxBS,OAAO;EDyBrB,gBAAgB,ECvBA,OAAO;CD2B1B;;AA5BH,AAyBM,MAzBA,AAqBH,kBAAkB,AAId,MAAM,EAzBb,MAAM,AAsBH,iBAAiB,AAGb,MAAM,CAAC;EACN,gBAAgB,ECtBV,IAAI;CDuBX;;AA3BP,AA+BM,MA/BA,AA+BC,gBAAQ,EA/Bf,MAAM,AA+BC,kBAAQ,CAAC;EACR,aAAa,EAAE,KAAK;CACrB;;AAjCP,AAkCM,MAlCA,AAkCC,eAAO,EAlCd,MAAM,AAkCC,iBAAO,CAAC;EACP,aAAa,EAAE,GAAG;CACnB;;AApCP,AAuCE,MAvCI,AAuCH,aAAa,CAAC;EACb,gBAAgB,ECtCF,OAAO;EDuCrB,KAAK,EAAE,OAAO;EACd,aAAa,EAAE,GAAG;CACnB;;AA3CH,AA6CE,MA7CI,AA6CH,cAAc,CAAC;EACd,aAAa,EAAE,KAAK;EACpB,cAAc,EAAE,SAAS;EACzB,KAAK,EC5CK,IAAI;ED6Cd,UAAU,EAAE,wDAA0E;CACvF;;AAlDH,AAoDE,MApDI,AAoDH,MAAM,CAAC;EACN,SAAS,EAAE,IAAI;EACf,OAAO,EAAE,QAAQ;CAClB;;AAvDH,AAyDE,MAzDI,AAyDH,OAAO,CAAC;EACP,SAAS,EAAE,IAAI;CAChB",
    "sources": [
        "button.scss",
        "../../../variables.scss"
    ],
    "names": [],
    "file": "button.css"
}