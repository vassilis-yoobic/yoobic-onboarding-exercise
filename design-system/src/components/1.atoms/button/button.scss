@import "../../../variables.scss";

button {
  display: flex;
  align-items: center;
  padding: 20px;
  font-weight: 700;
  transition: all 0.15s ease 0s;
  text-transform: uppercase;
  letter-spacing: 0.56px;
  text-decoration: none;
  border: none;
  outline: none;
  cursor: pointer;

  &.Primary-Rounded,
  &.Primary-Square {
      color: $color-white;
      background-color: $primary-color-btn;
      &:hover {
        background-color: $primary-darker-btn;
      }
  }
  &.Secondary-Rounded,
  &.Secondary-Square {
      color: $primary-color-btn;
      background-color: $secondary-color-btn;
      &:hover {
        background-color: $color-white;
      }
  }
  &.Primary,
  &.Secondary {
      &-Rounded {
        border-radius: 100px;
      }
      &-Square {
        border-radius: 8px;
      }
  }

  &.Third-Square {
    background-color: $third-color-btn;
    color: #e0e0f1;
    border-radius: 8px;
  }

  &.With-Gradient {
    border-radius: 200px;
    text-transform: lowercase;
    color: $color-white;
    background: linear-gradient(89.97deg, rgb(83, 83, 255) 0.03%, rgb(255, 74, 85) 99.14%);
  }

  &.Small {
    font-size: 12px;
    padding: 6px 12px;
  }

  &.Medium {
    font-size: 14px;
  }
}

