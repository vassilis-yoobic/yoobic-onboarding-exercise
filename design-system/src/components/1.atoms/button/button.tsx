import { Component, ComponentInterface, EventEmitter, h, Host, Prop } from '@stencil/core';

import { Identifier } from '../../shared/identifier';

@Component({
  tag: 'my-btn',
  styleUrl: 'button.css',
})

export class MyBtn implements ComponentInterface {
  @Prop() identifier?: Identifier = 'button';
  @Prop() label?: string;
  @Prop() size: "Small" | "Medium";
  @Prop() type: 'Primary-Rounded' | 'Secondary-Rounded' | 'Primary-Square' | 'Secondary-Square' | 'Third-Square' | 'With-Gradient';

  buttonClicked: EventEmitter<Identifier>;

  buttonClickedHandler() {
    alert('Button Clicked')
  }

  render() {
    return (
      <Host>
        <button class={`button ${this.size} ${this.type}`} onClick={this.buttonClickedHandler.bind(this)}>
          {this.label && this.label}
          <slot />
        </button>
      </Host>);
  }
}
