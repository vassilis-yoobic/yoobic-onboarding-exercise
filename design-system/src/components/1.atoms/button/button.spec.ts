import { newSpecPage } from '@stencil/core/testing';

import { MyBtn } from './button';

describe('1.atoms/button', () => {
  describe('rendering', () => {
    let page;

    beforeEach(async () => {
      page = await newSpecPage({
        components: [MyBtn],
        html: '<my-btn>Test</my-btn>'
      });
    });
    describe('props', () => {
      it('should render a button with default attributes', async () => {
        expect(page.root).toMatchSnapshot();
      })
      it('should render a small button', async () => {
        page.root.size = 'small';
        await page.waitForChanges();
        expect(page.root).toMatchSnapshot();
      });
      it('should render a square button', async () => {
        page.root.type = 'Primary-Square';
        await page.waitForChanges();
        expect(page.root).toMatchSnapshot();
      });
    })
    describe('events', () => {
      it('should fire a click event when enabled', async () => {
        const onClick = jest.fn();
        page.root.addEventListener('click', onClick);
        await page.waitForChanges();
        page.root.click();
        expect(onClick).toHaveBeenCalled();
      });
    })
  });
});
