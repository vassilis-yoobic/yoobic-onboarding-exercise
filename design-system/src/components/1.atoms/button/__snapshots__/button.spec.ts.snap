// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1.atoms/button rendering props should render a button with default attributes 1`] = `
<my-btn>
  <!---->
  <button class="button undefined">
    Test
  </button>
</my-btn>
`;

exports[`1.atoms/button rendering props should render a small button 1`] = `
<my-btn>
  <!---->
  <button class="button small undefined">
    Test
  </button>
</my-btn>
`;

exports[`1.atoms/button rendering props should render a square button 1`] = `
<my-btn>
  <!---->
  <button class="Primary-Square button undefined">
    Test
  </button>
</my-btn>
`;
